test_plans/atomic_ref.asciidoc

```diff
@@ -505,4 +505,264 @@ Check if it returns the original value of the referenced object. Check returned
+* `float`
+* `long long` (if `sizeof(long long) == 8` then only if the device has `aspect::atomic64`)
+* `double` (only if the device has `aspect::fp64` and if `sizeof(double) == 8` only if the device has `aspect::atomic64`)
+
+These `SCOPE` values which are supported by the device (according to `info::device::atomic_memory_scope_capabilities`):
+
+* `memory_scope::device`
+* `memory_scope::system`
+* `memory_scope::work_group`
+
+These `ORDER` values which are supported by the device (according to `info::device::atomic_memory_order_capabilities`):
+
+* `memory_order::acq_rel`
+* `memory_order::seq_cst`
+
+These `SPACE` values:
+
+* `access::address_space::local_space`
+* `access::address_space::generic_space`
+
+==== Ordering
+
+This test uses a `parallel_for` with `nd_range` parameter to atomically read scalar in one thread,
+and then aquire all changes from it in another thread within local group.
+
+[source,c++]
+----
+size_t local_range = q.get_info<sycl::info::device::max_work_group_size>();
+sycl::nd_range nd_range(GLOBAL_RANGE, local_range);
+std::array<bool, GLOBAL_RANGE> res;
+vals.fill(false);
+sycl::buffer buf{res.data(), {GLOBAL_RANGE}};
+q.submit([&](sycl::handler &cgh) {
+  sycl::accessor res_acc{buf, cgh};
+  sycl::local_accessor<TYPE> local_acc{cgh};
+  sycl::local_accessor<bool> arr_acc{{local_range}, cgh};
+  cgh.parallel_for(nd_range, [=](auto item) {
+     sycl::atomic_ref<TYPE, ORDER, SCOPE, SPACE> a_r{local_acc[0]};
+     arr_acc[item.local_id()] = true;
+     a_r.store(item.local_id(), ORDER1);
+     res_acc[item.global_id()] = (arr_acc[a_r.load(ORDER2)] == true);
+  });
+});
+----
+
+Verify that all elements of res is `true` after the loop completes.
+
+This test is repeated for all combinations of supported values listed below:
+
+These `TYPE` types:
+
+* `int`
+* `float`
+* `long long` (if `sizeof(long long) == 8` then only if the device has `aspect::atomic64`)
+* `double` (only if the device has `aspect::fp64` and if `sizeof(double) == 8` only if the device has `aspect::atomic64`)
+
+These `SCOPE` values which are supported by the device (according to `info::device::atomic_memory_scope_capabilities`):
+
+* `memory_scope::device`
+* `memory_scope::system`
+* `memory_scope::work_group`
+
+These `ORDER` values which are supported by the device (according to `info::device::atomic_memory_order_capabilities`):
+
+* `memory_order::relaxed`
+* `memory_order::acq_rel`
+* `memory_order::seq_cst`
+
+These `ORDER1` and `ORDER2` values which are supported by the device (according to `info::device::atomic_memory_order_capabilities`):
+
+* `ORDER1` is `memory_order::release` and `ORDER2` is `memory_order::acquire`
+* `ORDER1` and `ORDER2` are both `memory_order::seq_cst`
+
+These `SPACE` values:
+
+* `access::address_space::local_space`
+* `access::address_space::generic_space`
+
+==== Atomicity with respect to atomic operations in host code
+
+This check is skipped if device doesn't support `aspect::usm_atomic_shared_allocations`.
```

Comment:
The device also needs to support `memory_scope::system` according to `info::device::atomic_memory_scope_capabilities`.

Reply:
Added note about skipping check if `memory_scope::system` is not supported.